src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java

```diff
@@ -7,37 +7,82 @@
 
 import meteordevelopment.meteorclient.events.world.TickEvent;
 import meteordevelopment.meteorclient.settings.BoolSetting;
+import meteordevelopment.meteorclient.settings.EnumSetting;
 import meteordevelopment.meteorclient.settings.Setting;
 import meteordevelopment.meteorclient.settings.SettingGroup;
 import meteordevelopment.meteorclient.systems.modules.Categories;
 import meteordevelopment.meteorclient.systems.modules.Module;
+import meteordevelopment.meteorclient.utils.player.PlayerUtils;
 import meteordevelopment.orbit.EventHandler;
 
 public class Sprint extends Module {
     private final SettingGroup sgGeneral = settings.getDefaultGroup();
 
-    private final Setting<Boolean> whenStationary = sgGeneral.add(new BoolSetting.Builder()
-        .name("when-stationary")
-        .description("Continues sprinting even if you do not move.")
+    public enum Mode {
+        Strict,
+        Rage,
+        Omni
+    }
+
+    private final Setting<Mode> mode = sgGeneral.add(new EnumSetting.Builder<Mode>()
+        .name("speed-mode")
+        .description("What mode of sprinting.")
+        .defaultValue(Mode.Strict)
+        .build()
+    );
+
+    // Removed whenStationary as it was just Rage sprint
+    private final Setting<Boolean> hunger = sgGeneral.add(new BoolSetting.Builder()
+        .name("check-for-hunger")
+        .description("Checks if you have enough hunger to sprint")
         .defaultValue(false)
         .build()
     );
 
+
     public Sprint() {
         super(Categories.Movement, "sprint", "Automatically sprints.");
     }
 
+    public Mode currentmode;
+
     @Override
     public void onDeactivate() {
         mc.player.setSprinting(false);
     }
 
     @EventHandler
     private void onTick(TickEvent.Post event) {
-        if (mc.player.forwardSpeed > 0 && !whenStationary.get()) {
-            mc.player.setSprinting(true);
-        } else if (whenStationary.get()) {
-            mc.player.setSprinting(true);
+        currentmode = mode.get();
```

Comment:
why is this even a thing, just do switch (mode.get()) { ... }
the variable shouldn't exist